src/logic/__tests__/startAccountMigration.test.js

```diff
@@ -0,0 +1,137 @@
+import {Observable} from 'rxjs';
+import reduce from 'lodash-es/reduce';
+
+import reducer from '../../reducers/user';
+import startAccountMigration from '../startAccountMigration';
+import {
+  accountMigrationComplete,
+  accountMigrationError,
+  accountMigrationNeeded,
+  accountMigrationUndoPeriodExpired,
+  startAccountMigration as startAccountMigrationAction,
+  userAuthenticated,
+  dismissAccountMigration,
+} from '../../actions/user';
+import {migrateAccount} from '../../clients/firebase';
+import {getCurrentAccountMigration} from '../../selectors';
+import {bugsnagClient} from '../../util/bugsnag';
+
+import {
+  credentialFactory,
+  firebaseErrorFactory,
+  userFactory,
+} from '@factories/clients/firebase';
+import {githubProfileFactory} from '@factories/clients/github';
+import {firebaseProjectFactory} from '@factories/data/firebase';
+
+jest.mock('../../clients/firebase.js');
+jest.mock('../../selectors');
+jest.mock('../../util/bugsnag.js');
+jest.useFakeTimers();
+
+describe('startAccountMigration', () => {
+  const dispatch = jest.fn();
+  const done = jest.fn();
+
+  test('not dismissed during undo period, successful migration', async () => {
+    const mockUser = userFactory.build();
+    const mockCredential = credentialFactory.build();
+    const mockProfile = githubProfileFactory.build();
+    const mockProjects = [
+      firebaseProjectFactory.build(),
+      firebaseProjectFactory.build(),
+    ];
```

Comment:
Small thing, but you can use the [`buildList()`](https://github.com/rosiejs/rosie#buildlist) function to avoid having to manually create multiple instances